File: src/dotnet/HQ.Abstractions/Invoices/UpdateInvoiceV1.cs
namespace HQ.Abstractions.Invoices
{
    public class UpdateInvoiceV1
    {
        public class Request
        {
            public Guid Id { get; set; }
            public Guid? ClientId { get; set; }
            public DateOnly? Date { get; set; }
            public string? InvoiceNumber { get; set; }
        }

        public class Response
        {
            public Guid Id { get; set; }
            public Guid ClientId { get; set; }
            public string ClientName { get; set; } = null!;
            public DateOnly Date { get; set; }
            public string InvoiceNumber { get; set; } = null!;
            public decimal Total { get; set; }
            public decimal TotalApprovedHours { get; set; }
            public int TimeEntriesCount { get; set; }
        }
    }
}

File: src/angular/hq/src/app/models/Invoices/update-invoice-v1.ts
export interface UpdateInvoiceRequestV1 {
  id?: string;
  date: string | null;
  invoiceNumber: string | null;
  total: number | null;
  totalApprovedHours: number | null;
}

export interface UpdateInvoiceResponseV1 {
  id: string;
}
